### DISCUSSION TITLE
Prepend comments to COMMIT statements generated by SQLAlchemy sessions

### DISCUSSION THREAD

**Discussion by tonycosentini** [sqlalchemy/sqlalchemy#10797] [Usage Questions]
Is there any way to prepend a comment to a `COMMIT` statement generated from calling `commit()` on a session?

When using AWS RDS, there is a nice dashboard to show the amount of load generated by different queries, but if you're using SQL transactions, all the writes will get bundled up into a single line with just `COMMIT`. I'd like to prepend comments to some of the higher QPS SQL transactions in our codebase so I can differentiate them on this dashboard.

Ideally I'd be able to give a custom label at each commit/SQL transaction to differentiate them.

edit: Following up on this, I'm going to guess that it's not really possible since the commit itself is handled entirely by whatever is managing the DB connection (in my pass psycopg2).

**Comment by zzzeek** [comment]
SQLAlchemy doesnt emit "COMMIT" as a SQL statement, it calls `connection.commit()` on the DBAPI connection.  The logging you see that says "COMMIT" is simply hardcoded to `log.info("COMMIT")`.

if you want to add your own logging for transaction commits you have a choice of events to do so, including [ConnectionEvents.commit](https://docs.sqlalchemy.org/en/20/core/events.html#sqlalchemy.events.ConnectionEvents.commit) for engine level, [SessionEvents.before_commit](https://docs.sqlalchemy.org/en/20/orm/events.html#sqlalchemy.orm.SessionEvents.before_commit) for Session-level, per connection, and [SessionEvents.after_transaction_end](https://docs.sqlalchemy.org/en/20/orm/events.html#sqlalchemy.orm.SessionEvents.after_transaction_end), end of the logical transaction within the `Session`.

if you use the `sqlalchemy.engine` logger name your log lines would be emitted on the same logger.
